src/neo-vm/ExecutionEngine.cs

```diff
@@ -1193,15 +992,44 @@ protected virtual void LoadContext(ExecutionContext context)
 
         public ExecutionContext LoadScript(Script script, int rvcount = -1)
         {
-            ExecutionContext context = new ExecutionContext(script, CurrentContext?.Script, rvcount);
+            ExecutionContext context = new ExecutionContext(script, CurrentContext?.Script, rvcount, ReferenceCounter);
             LoadContext(context);
             return context;
         }
 
         protected virtual bool OnSysCall(uint method) => false;
 
-        protected virtual bool PostExecuteInstruction(Instruction instruction) => true;
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public StackItem Peek(int index = 0)
+        {
+            var stack = CurrentContext?.EvaluationStack ?? ResultStack;
```

Comment:
This could be confuse, can we use different methods for different stacks?

Reply:
`ResultStack` is only available after the VM has finished running. At this time, it can be considered as the current stack, so there is no confusion.

Reply:
But we are doing extra work for the opcodes execution.

Reply:
You are right. I will remove the `ResultStack` check.